internal/dispatch/keys/computed_test.go

```diff
@@ -117,7 +117,35 @@ func TestStableCacheKeys(t *testing.T) {
 					},
 				}, computeBothHashes)
 			},
-			"87b8e4dcf893f4abd701",
+			"b4989ce7d2f695c251",
+		},
+		{
+			"lookup resources with zero limit",
+			func() DispatchCacheKey {
+				return lookupResourcesRequestToKey(&v1.DispatchLookupResourcesRequest{
+					ObjectRelation: RR("document", "view"),
+					Subject:        ONR("user", "mariah", "..."),
+					Metadata: &v1.ResolverMeta{
+						AtRevision: "1234",
+					},
+					OptionalLimit: 0,
+				}, computeBothHashes)
+			},
+			"b4989ce7d2f695c251",
+		},
+		{
+			"lookup resources with non-zero limit",
+			func() DispatchCacheKey {
+				return lookupResourcesRequestToKey(&v1.DispatchLookupResourcesRequest{
+					ObjectRelation: RR("document", "view"),
+					Subject:        ONR("user", "mariah", "..."),
+					Metadata: &v1.ResolverMeta{
+						AtRevision: "1234",
+					},
+					OptionalLimit: 42,
+				}, computeBothHashes)
+			},
+			"a1bcf8c7e581fb9be401",
```

Comment:
We could try to be smart about this, in accepting cache entries whose limits are higher than the user specifies, but realistically, it will cover a very small case. Longer term, once we support default pagination, this should be less of an issue